File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/methodadapter/tree/MethodProxyAdapter.java
package net.yakclient.mixin.base.internal.methodadapter.tree;

import net.yakclient.mixin.base.internal.bytecode.ByteCodeUtils;
import org.objectweb.asm.MethodVisitor;
import org.objectweb.asm.tree.MethodNode;

public class MethodProxyAdapter extends MethodVisitor {
    private final MethodVisitor parent;

    public MethodProxyAdapter(ASMMethodProxy proxy, MethodVisitor parent) {
        super(ByteCodeUtils.ASM_VERSION, proxy);
        this.parent = parent;
    }

    @Override
    public void visitEnd() {
        ((MethodNode) this.mv).accept(this.parent);
    }
}

File: mixin-base/src/main/kotlin/net/yakclient/mixin/base/internal/methodadapter/tree/MethodProxyAdapter.kt
package net.yakclient.mixin.base.internal.methodadapter.tree

import net.yakclient.mixin.base.internal.bytecode.ByteCodeUtils
import org.objectweb.asm.MethodVisitor
import org.objectweb.asm.tree.MethodNode

class MethodProxyAdapter(
    proxy: TreeMethodProxy,
    private val parent: MethodVisitor
) : MethodVisitor(ByteCodeUtils.ASM_VERSION, proxy) {

    override fun visitEnd() {
        super.visitEnd()
        (mv as MethodNode).accept(parent)
    }
}